### PR TITLE
Add boardsource/the_q

### PR DESCRIPTION
**Pull request by waffle87**
## Description

Adds v3 definition of Boardsouce The Q keyboard

## QMK Pull Request

[qmk/qmk_firmware:23782](https://github.com/qmk/qmk_firmware/pull/23782)

## Checklist

- [x] The VIA support for this keyboard is **MERGED** in QMK master already **(MANDATORY)**
- [x] The VIA definition follows the guide here: https://caniusevia.com/docs/layouts
- [x] I have a V3 JSON version for this keyboard definition.**(MANDATORY)**
- [x] I have formatted the JSON file to have consistent formatting with the rest of the repository.
- [x] I have tested this keyboard definition using VIA's "Design" tab.
- [x] I have tested this keyboard definition with firmware on a device.
- [x] I have assigned alpha keys and modifier keys with the correct colors.
- [x] The Vendor ID is not `0xFEED`

### CODE REVIEW

File: v3/boardsource/the_q/the_q.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "The Q",
+  "vendorId": "0x4273",
+  "productId": "0x7552",
+  "matrix": {"rows": 7, "cols": 6},
+  "keycodes": ["qmk_lighting"],
+  "menus": ["qmk_rgb_matrix"],
```
Comment (Xelus22) [inline v3/boardsource/the_q/the_q.json:7]: This enables all animations, just making sure thats what you want? QMK source only enables a handful it looks like, would require a custom menu to show the correct amount

Comment (waffle87) [reply v3/boardsource/the_q/the_q.json]: Sorry this was forgotten, fixed now.

Comment (Cipulot) [reply v3/boardsource/the_q/the_q.json]: To have the correct options shown, you should add the custom menu RGB Matrix explicit declaration and only use the effects you have enabled by removing the unused ones from the option list: https://www.caniusevia.com/docs/built_in_menus#qmk_rgb_matrix

Comment (waffle87) [reply v3/boardsource/the_q/the_q.json]: Understood, thanks for the pointer.

**Comment by Cipulot** [comment]
Keymap mode after QMK breaking change here: https://github.com/the-via/qmk_userspace_via/tree/main/keyboards/boardsource/the_q

File: v3/boardsource/the_q/the_q.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      [{"y": -0.87,"x": 5},"2,5",{"x": 3.5},"4,5"],
+      [{"y": -0.87},"2,0","2,1",{"x": 11.5},"4,1","4,0"],
+      [{"y": -0.129,"x": 3.5},"3,0",{"x": 6.5},"3,5"],
+      [{"r": 15,"rx": 3.75,"ry": 3.25,"y": -0.2018,"x": 1},"3,1"],
+      [{"r": 30,"rx": 5.4,"ry": 9.3,"y": -5.64,"x": -2.2802,"h": 1.5},"3,2"],
+      [{"r": -30,"rx": 11.1,"y": -6.12,"x": 0.4193,"h": 1.5},"3,3"],
+      [{"r": -15,"rx": 12,"ry": 9.1,"y": -5.92,"x": -0.7304},"3,4"]
+    ]
+  }
+}
```
Comment (Cipulot) [inline v3/boardsource/the_q/the_q.json:74]:
```suggestion
{
  "name": "The Q",
  "vendorId": "0x4273",
  "productId": "0x7552",
  "matrix": {
    "rows": 7,
    "cols": 6
  },
  "keycodes": [
    "qmk_lighting"
  ],
  "menus": [
    {
      "label": "Lighting",
      "content": [
        {
          "label": "Backlight",
          "content": [
            {
              "label": "Brightness",
              "type": "range",
              "options": [
                0,
                255
              ],
              "content": [
                "id_qmk_rgb_matrix_brightness",
                3,
                1
              ]
            },
            {
              "label": "Effect",
              "type": "dropdown",
              "content": [
                "id_qmk_rgb_matrix_effect",
                3,
                2
              ],
              "options": [
                "All Off",
                "Solid Color",
                "Alphas Mods",
                "Band Sat.",
                "Band Val.",
                "Breathing",
                "Gradient Left/Right",
                "Gradient Up/Down"
              ]
            },
            {
              "showIf": "{id_qmk_rgb_matrix_effect} != 0",
              "label": "Effect Speed",
              "type": "range",
              "options": [
                0,
                255
              ],
              "content": [
                "id_qmk_rgb_matrix_effect_speed",
                3,
                3
              ]
            },
            {
              "showIf": "{id_qmk_rgb_matrix_effect} != 0 && {id_qmk_rgb_matrix_effect} != 24 && {id_qmk_rgb_matrix_effect} != 28 && {id_qmk_rgb_matrix_effect} != 29 && {id_qmk_rgb_matrix_effect} != 32",
              "label": "Color",
              "type": "color",
              "content": [
                "id_qmk_rgb_matrix_color",
                3,
                4
              ]
            }
          ]
        }
      ]
    }
  ],
  "layouts": {
    "keymap": [
      [
        {
          "x": 3
        },
        "0,3",
        {
          "x": 7.5
        },
        "6,3"
      ],
      [
        {
          "y": -0.87,
          "x": 2
        },
        "0,2",
        {
          "x": 1
        },
        "0,4",
        {
          "x": 5.5
        },
        "6,4",
        {
          "x": 1
        },
        "6,2"
      ],
      [
        {
          "y": -0.87,
          "x": 5
        },
        "0,5",
        {
          "x": 3.5
        },
        "6,5"
      ],
      [
        {
          "y": -0.87
        },
        "0,0",
        "0,1",
        {
          "x": 11.5
        },
        "6,1",
        "6,0"
      ],
      [
        {
          "y": -0.379,
          "x": 3
        },
        "1,3",
        {
          "x": 7.5
        },
        "5,3"
      ],
      [
        {
          "y": -0.87,
          "x": 2
        },
        "1,2",
        {
          "x": 1
        },
        "1,4",
        {
          "x": 5.5
        },
        "5,4",
        {
          "x": 1
        },
        "5,2"
      ],
      [
        {
          "y": -0.87,
          "x": 5
        },
        "1,5",
        {
          "x": 3.5
        },
        "5,5"
      ],
      [
        {
          "y": -0.87
        },
        "1,0",
        "1,1",
        {
          "x": 11.5
        },
        "5,1",
        "5,0"
      ],
      [
        {
          "y": -0.379,
          "x": 3
        },
        "2,3",
        {
          "x": 7.5
        },
        "4,3"
      ],
      [
        {
          "y": -0.87,
          "x": 2
        },
        "2,2",
        {
          "x": 1
        },
        "2,4",
        {
          "x": 5.5
        },
        "4,4",
        {
          "x": 1
        },
        "4,2"
      ],
      [
        {
          "y": -0.87,
          "x": 5
        },
        "2,5",
        {
          "x": 3.5
        },
        "4,5"
      ],
      [
        {
          "y": -0.87
        },
        "2,0",
        "2,1",
        {
          "x": 11.5
        },
        "4,1",
        "4,0"
      ],
      [
        {
          "y": -0.129,
          "x": 3.5
        },
        "3,0",
        {
          "x": 6.5
        },
        "3,5"
      ],
      [
        {
          "r": 15,
          "rx": 3.75,
          "ry": 3.25,
          "y": -0.2018,
          "x": 1
        },
        "3,1"
      ],
      [
        {
          "r": 30,
          "rx": 5.4,
          "ry": 9.3,
          "y": -5.64,
          "x": -2.2802,
          "h": 1.5
        },
        "3,2"
      ],
      [
        {
          "r": -30,
          "rx": 11.1,
          "y": -6.12,
          "x": 0.4193,
          "h": 1.5
        },
        "3,3"
      ],
      [
        {
          "r": -15,
          "rx": 12,
          "ry": 9.1,
          "y": -5.92,
          "x": -0.7304
        },
        "3,4"
      ]
    ]
  }
}

```

**Review by Cipulot — APPROVED** [review]

File: v3/boardsource/the_q/the_q.json (hunk 2 above)
Comment (Cipulot) [reply v3/boardsource/the_q/the_q.json:74]: Simple formatting.

**Pull request merged**